src/core.js

```diff
@@ -338,8 +338,34 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
      * @param {boolean} [keepEmptyRows] Optional. Flag for preventing deletion of empty rows.
      */
     alter(action, index, amount = 1, source, keepEmptyRows) {
-      let delta;
+      const trimSelectionToTableSize = () => {
+        const currentSelectedRange = selection.selectedRange.current();
+        const currentRowCount = instance.countRows();
+        const currentColumnCount = instance.countCols();
+
+        selection.getSelectedRange().pop();
+
+        selection.setRangeStartOnly(
+          new CellCoords(currentSelectedRange.from.row, currentSelectedRange.from.col),
+          true
+        );
+
+        if (selection.isSelectedByCorner()) {
+          selection.selectedByRowHeader.add(selection.getLayerLevel());
```

Comment:
I think It's a bad idea to modify the internal props (like `selectedByRowHeader`) of the Selection class from the outside. Can you move the logic to the Selection class?